jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRolesViewImpl.java

```diff
@@ -129,23 +157,21 @@ public void setupPagination() {
     public void setVisibleItems(List visibleItems) {
         removeAllChildren(roles);
         int visibleItemsSize = visibleItems.size();
-        if(visibleItemsSize>0){
-            visibleItems.stream().forEach(e -> ((CaseRoleItemView)e).setLastElementStyle(false));
-            ((CaseRoleItemView)visibleItems.get(visibleItemsSize-1)).setLastElementStyle(true);
+        if (visibleItemsSize > 0) {
+            visibleItems.forEach(e -> ((CaseRoleItemView) e).setLastElementStyle(false));
+            ((CaseRoleItemView) visibleItems.get(visibleItemsSize - 1)).setLastElementStyle(true);
         }
-        visibleItems.stream().forEach(e -> roles.appendChild(((CaseRoleItemView)e).getElement()) );
+        visibleItems.forEach(e -> roles.appendChild(((CaseRoleItemView) e).getElement()));
     }
 
     @Override
     public Div getScrollBox() {
         return scrollbox;
     }
 
-    @EventHandler("user-add")
-    public void onUserAddClick(@ForEvent("click") Event e) {
-        if (userAddCommand != null) {
-            userAddCommand.execute();
-        }
+    @EventHandler("filter-select")
+    public void onRolesAssignmentFilterChange(@ForEvent("change") Event e) {
+        filterCommand.execute();
```

Comment:
@bsremac  this event is fired when the filter select value changes, With the current implementation, yes, the list of roles assignment is requested again. Let me discuss with @cristianonicolai if we can store the values in any array at the presenter and avoid to reload it every time.